onelogin_sync/onelogin_sync.py

```diff
@@ -0,0 +1,97 @@
+#! /usr/bin/python3
+
+import os
+import sys
+import logging
+import re
+import configargparse
+import boto3
+
+from onelogin.api.client import OneLoginClient
+
+
+def get_opts(log):
+    parser = configargparse.ArgParser(default_config_files=['onelogin_lookup.ini'])
+    parser.add('-v', '--verbosity', help='Logging level. Default ERROR', default='error')
+    parser.add('-c', '--config', help='Config file')
+
+    opts = parser.parse_args()
+
+    if opts.verbosity.lower() == 'debug':
+        log.setLevel(logging.DEBUG)
+    elif opts.verbosity.lower() == 'info':
+        log.setLevel(logging.INFO)
+    elif opts.verbosity.lower() == 'warning':
+        log.setLevel(logging.WARNING)
+    elif opts.verbosity.lower() == 'error':
+        log.setLevel(logging.ERROR)
+    elif opts.verbosity.lower() == 'critical':
+        log.setLevel(logging.CRITICAL)
+    return opts
+
+
+def get_logger():
+    logging.basicConfig(format = '%(asctime)s %(levelname)-8s %(message)s',
+                        datefmt='%Y-%m-%d %H:%M:%S')
+    log = logging.getLogger('default')
+    return log
+
+
+def setup():
+    log = get_logger()
+    opts = get_opts(log)
+    return opts, log
+
+
+def get_onelogin_credentials():
+    ssm = boto3.client('ssm')
+    parameter = ssm.get_parameter(Name='/bastion/onelogin_id', WithDecryption=True)
+    key_id = parameter['Parameter']['Value']
+    parameter = ssm.get_parameter(Name='/bastion/onelogin_secret', WithDecryption=True)
+    key_secret = parameter['Parameter']['Value']
+    return key_id, key_secret
+
+
+def get_user_list(key_id, key_secret, log):
+    try:
+        onelogin_client = OneLoginClient(
+            key_id,
+            key_secret,
+            "us"
+        )
+    except Exception as onelogin_exception:
+        log.error(onelogin_exception)
+        sys.exit(1)
+
+    onelogin_users = onelogin_client.get_users()
+
+    return onelogin_users
+
+
+def create_delete_users(log, users=None):
+    os.environ['PATH'] += os.pathsep + '/usr/sbin'
+    for u in users:
+
+        if not re.match('^[a-z][-a-z0-9]*$', u.username):
+            log.warning('Skipping invalid username %s' % u.username)
+            continue
+
+        if u.status in [1, 3, 4] and u.state == 1:
```

Comment:
All active OL users with a key registered are granted access to theses bastions.  Is this tenable for the short term?  I  suspect we will want some level of per-bastion access control like implemented in the [onelogin_users Ansible module](https://github.com/PatientPing/sysadmin/blob/master/ansible-pull/roles-include/onelogin_users/defaults/main.yml#L3)

Reply:
Added support for filtering by role in 0f1751e0484f42951da7bb652a9a7493ceeebde1.  Look ok @trobinsonpp ?

Reply:
Looks all set() :)